src/index.js

```diff
@@ -71,37 +70,22 @@ export function withOptions(handler, options) {
   };
 }
 
-class EventListener extends React.Component {
+class EventListener extends React.PureComponent {
   componentDidMount() {
-    this.addListeners();
-  }
-
-  shouldComponentUpdate(nextProps) {
-    return !shallowEqual(this.props, nextProps);
-  }
-
-  componentWillUpdate() {
-    this.removeListeners();
-  }
-
-  componentDidUpdate() {
-    this.addListeners();
-  }
-
-  componentWillUnmount() {
-    this.removeListeners();
+    this.applyListeners(on);
   }
 
-  addListeners() {
+  componentDidUpdate(prevProps) {
+    this.applyListeners(off, prevProps);
     this.applyListeners(on);
   }
 
-  removeListeners() {
+  componentWillUnmount() {
     this.applyListeners(off);
   }
 
-  applyListeners(onOrOff) {
-    const { target } = this.props;
+  applyListeners(onOrOff, props = this.props) {
```

Comment:
Why this?

Reply:
When the component update, we remove the old listeners and add the new ones.